src/function/utility/typeof.cpp

```diff
@@ -0,0 +1,42 @@
+#include "function/scalar_function.h"
+#include "function/utility/function_string_bind_data.h"
+#include "function/utility/scalar_utility_functions.h"
+
+using namespace kuzu::common;
+
+namespace kuzu {
+namespace function {
+
+static std::unique_ptr<FunctionBindData> bindFunc(const binder::expression_vector& arguments,
+    Function* /*function*/) {
+    std::unique_ptr<FunctionBindData> bindData;
+    if (arguments[0]->getDataType().getLogicalTypeID() == LogicalTypeID::ANY) {
+        bindData = std::make_unique<FunctionStringBindData>("NULL");
+        bindData->paramTypes.push_back(*LogicalType::STRING());
+    } else {
+        bindData = std::make_unique<FunctionStringBindData>(arguments[0]->getDataType().toString());
+    }
+    return bindData;
+}
+
+static void execFunc(const std::vector<std::shared_ptr<ValueVector>>& params, ValueVector& result,
+    void* dataPtr) {
+    KU_ASSERT(params.size() == 1);
+    result.resetAuxiliaryBuffer();
+    auto typeData = reinterpret_cast<FunctionStringBindData*>(dataPtr);
```

Comment:
Use dataPtr->constPtrCast()

Reply:
I think we cannot use ```constPtrCast()``` now since we are casting from ```void*```. This should be fixed after we refactor execFunc according to [Xiyang's suggestion](https://github.com/kuzudb/kuzu/pull/3282#discussion_r1578819591).